lua/luasnip/extras/snippet_list.lua

```diff
@@ -0,0 +1,92 @@
+local function get_context(snippet)
+	return {
+		name = snippet.name,
+		trigger = snippet.trigger,
+		description = snippet.dscr,
+		wordTrig = snippet.wordTrig and true or false,
+		regTrig = snippet.regTrig and true or false,
+		docstring = snippet:get_docstring(),
+	}
+end
+
+local function get_name(bufnr)
+	return "LuaSnip://Snippets buf " .. bufnr
+end
+
+local function printer(snippets)
+	return vim.inspect(snippets)
+end
+
+local win_opts = { foldmethod = "indent" }
+local buf_opts = { filetype = "lua" }
+
+local function set_win_opts(win, opts)
+	for opt, val in pairs(opts) do
+		vim.api.nvim_win_set_option(win, opt, val)
+	end
+end
+
+local function set_buf_opts(buf, opts)
+	for opt, val in pairs(opts) do
+		vim.api.nvim_buf_set_option(buf, opt, val)
+	end
+end
+
+local function make_scratch_buf(buf)
+	local opts = {
+		buftype = "nofile",
+		bufhidden = "wipe",
+		buflisted = false,
+		swapfile = false,
+		modified = false,
+		modeline = false,
+	}
+
+	for opt, val in pairs(opts) do
+		vim.api.nvim_buf_set_option(buf, opt, val)
+	end
+end
+
+local function open_snippet_list(opts)
+	opts = opts or {}
+	opts.get_name = opts.get_name or get_name
+	opts.printer = opts.printer or printer
+	opts.win_opts = opts.win_opts or win_opts
+	opts.buf_opts = opts.buf_opts or buf_opts
+
+	-- load snippets before changing windows/buffers
+	local available = require("luasnip").available
```

Comment:
How about putting this outside the function, at the top of the file?

Reply:
Solved in https://github.com/L3MON4D3/LuaSnip/pull/652/commits/8cd564027f5fba9ab6d2af51da010dc051f71306